src/credentials/GenerateChipX509Cert.cpp

```diff
@@ -251,14 +258,18 @@ CHIP_ERROR EncodeExtensions(bool isCA, const Crypto::P256PublicKey & SKI, const
     {
         ASN1_START_SEQUENCE
         {
-            if (isCA)
+            if ((certType == CertType::kICA) || (certType == CertType::kRoot))
             {
                 ReturnErrorOnFailure(EncodeCASpecificExtensions(writer));
             }
-            else
+            else if (certType == CertType::kNode)
             {
                 ReturnErrorOnFailure(EncodeNOCSpecificExtensions(writer));
             }
+            else if (certType == CertType::kVidVerificationSigner)
+            {
+                ReturnErrorOnFailure(EncodeVendorIdVerificationSignerSpecificExtensions(writer));
+            }
```

Comment:
previous code used to `else` on all cases. Now we have everything fully coded in if/elseif so there is no "default" anymore if certType is some crazy type.
Would it make sense to use a `switch` for the compiler to enforce all branches taken or have an error branch with a error handle (e.g. log an error)?

Reply:
Done.